pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -77,6 +76,40 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if subnetPort.ObjectMeta.DeletionTimestamp.IsZero() {
+		if len(subnetPort.Spec.Subnet) != 0 {
```

Comment:
Do you think we can leverage logic in GetSubnetPathForSubnetPort
https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/subnetport/subnetport_controller.go#L408